backend/api/pdf_progress.py

```
@@ -0,0 +1,289 @@
+            except Exception as e:
+                logger.warning(f"Failed to store progress data in Redis: {e}")
+    
+    def _get_progress_data(self, session_id: str) -> Dict[str, Any]:
+        """Get progress data from Redis if available"""
+        if self.use_redis:
+            try:
+                data = self.redis.get(self._get_redis_key(session_id))
+                if data:
+                    # RedisManager already deserializes JSON, so return directly
+                    return data
+            except Exception as e:
```

Ruff (0.14.0)

Do not catch blind exception: `Exception`

(BLE001)